### PR TITLE
fix bpv bug

### PR DESCRIPTION
**Pull request by ywrsusan**
*Issue number of the reported bug or feature request: #<2337>*
https://jira4.prod.bloomberg.com/browse/RDISCROWD-2337

**Describe your changes**
check if value exists, if not, remove the key field

### CODE REVIEW

File: pybossa/view/projects.py [hunk 1]
```diff
@@ -3095,6 +3095,8 @@ def update_ext_config_dict(config):
         for _, fields in six.iteritems(ext_config):
             for key, value in six.iteritems(fields):
                 config[key] = value
```
Comment (dchhabda) [inline pybossa/view/projects.py]: value could be set when exist.this will avoid pop later

File: pybossa/view/projects.py [hunk 2]
```diff
@@ -3169,6 +3171,8 @@ def ext_config(short_name):
                     flash(gettext('Please correct the errors', 'error'))
                 ext_conf[form_name] = form.data
                 ext_conf[form_name].pop('csrf_token', None)     #Fflask-wtf v0.14.2 issue 102
+                if not ext_conf[form_name].get('bpv_key_id'):
+                    ext_conf[form_name].pop('bpv_key_id', None)
```
Comment (dchhabda) [inline pybossa/view/projects.py]: if condition not required.

Comment (ywrsusan) [reply pybossa/view/projects.py]: I think we only pop out the `bpv_key_id ` field when the value is `""`.

Comment (dchhabda) [reply pybossa/view/projects.py]: ok

File: pybossa/view/projects.py [hunk 3]
```diff
@@ -3169,6 +3168,8 @@ def ext_config(short_name):
                     flash(gettext('Please correct the errors', 'error'))
                 ext_conf[form_name] = form.data
                 ext_conf[form_name].pop('csrf_token', None)     #Fflask-wtf v0.14.2 issue 102
+                if not ext_conf[form_name].get('bpv_key_id'):
```
Comment (fiorda) [inline pybossa/view/projects.py]: Hardcoding an instance specific parameter here. Is there a way to make this more generic and prevent the same issue for other fields in the future?

Comment (ywrsusan) [reply pybossa/view/projects.py]: Yes I think it's doable, let me investigate a little bit.

**Pull request merged**